src/main.rs

```diff
@@ -61,23 +64,59 @@ pub fn run_command(ctx: &Context, command: Command) -> Result<()> {
     for err in &warnings {
         ctx.log_error_as_warning(err);
     }
+
+    // Release file mutex
+    match file_mutex
+        .map(|file| {
+            file.unlock()
+                .context("failed to release lock on config directory")
+        })
+        .transpose()
+    {
+        Ok(_) => {}
+        Err(err) => ctx.log_error_as_warning(&err),
+    };
+
     result
 }
 
-fn acquire_mutex(ctx: &Context, path: &Path) -> Result<fmutex::Guard> {
-    match fmutex::try_lock(path).with_context(|| format!("failed to open `{}`", path.display()))? {
-        Some(g) => Ok(g),
-        None => {
-            ctx.log_warning(
-                "Blocking",
-                &format!(
-                    "waiting for file lock on {}",
-                    ctx.replace_home(path).display()
-                ),
-            );
-            fmutex::lock(path)
-                .with_context(|| format!("failed to acquire file lock `{}`", path.display()))
-        }
+/// Returns the file to use for the filesystem mutex. On Windows, this is usually a lock
+/// file within the config directory.
+#[cfg(target_os = "windows")]
+fn get_file_for_mutex(path: &Path) -> io::Result<fs::File> {
```

Comment:
I would suggest adding a method on `Context` to return the `Path` for the file lock.